sdk/src/wallet/operations/transaction/high_level/allot_mana.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    client::{api::PreparedTransactionData, secret::SecretManage},
+    types::block::mana::ManaAllotment,
+    wallet::{
+        operations::transaction::{TransactionOptions, TransactionWithMetadata},
+        Wallet,
+    },
+};
+
+impl<S: 'static + SecretManage> Wallet<S>
+where
+    crate::wallet::Error: From<S::Error>,
+    crate::client::Error: From<S::Error>,
+{
+    pub async fn allot_mana(
+        &self,
+        allotments: impl IntoIterator<Item = impl Into<ManaAllotment>> + Send,
+        options: impl Into<Option<TransactionOptions>> + Send,
+    ) -> crate::wallet::Result<TransactionWithMetadata> {
+        let options = options.into();
+        let prepared_transaction = self.prepare_allot_mana(allotments, options.clone()).await?;
+
+        self.sign_and_submit_transaction(prepared_transaction, None, options)
+            .await
+    }
+
+    pub async fn prepare_allot_mana(
+        &self,
+        allotments: impl IntoIterator<Item = impl Into<ManaAllotment>> + Send,
```

Comment:
Why do we have this + the mana_allotments in the TransactionOptions?

Reply:
Yeah it may seem a little bit weird but it's for consistency. Allotments can already be set through the options so we need to take them into account.